src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,142 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T value) {
+            this.value = value;
+        }
+    }
+
+    public MyLinkedList() {
+
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (head == null) {
+            tail = head = newNode;
+        } else {
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode = new Node<>(value);
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Invalid index " + index);
+        }
+        if (head == null) {
+            head = tail = newNode;
+        } else if (index == 0) {
+            newNode.next = head;
+            head.prev = newNode;
+            head = newNode;
+        } else if (index == size) {
+            newNode.prev = tail;
+            tail.next = newNode;
+            tail = newNode;
+        } else {
+            Node<T> prev = getNodeByIndex(index - 1);
+            newNode.prev = prev;
+            newNode.next = prev.next;
+            prev.next = newNode;
+        }
+        size++;
```

Comment:
Here is a simpler algorithm to follow:

* If the index equals to size, treat it as an append operation and call the add(value) method directly.
* Check the index to ensure it's valid.
* Retrieve the node currently at the specified index. This will become the "next" node for the new node.
* Create a new node with its next reference pointing to the retrieved node and its prev reference pointing to the prev reference of the retrieved node.
* If the prev reference of the new node is null, it means we are inserting at the head, so update the first reference to point to the new node. Otherwise, update the next reference of the prev node to point to the new node.
* Finally, update the prev reference of the next node (the node retrieved initially) to point to the new node.
* Remember to increment the size.
This way, your code will be more concise and to the point.